codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/generators/EnumGenerator.kt

```diff
@@ -92,7 +94,12 @@ open class EnumGenerator(
 ) {
     protected val symbol: Symbol = symbolProvider.toSymbol(shape)
     protected val enumName: String = symbol.name
-    protected val meta = symbol.expectRustMetadata()
+    private val isSensitive = shape.shouldRedact(model)
+    protected val meta = if (isSensitive) {
+        symbol.expectRustMetadata().withoutDerives(RuntimeType.Debug)
```

Comment:
I think this should be done centrally in the `SymbolMetadataProvider`, where we can do the change for all applicable shapes: if they have the `sensitive` trait, remove `Debug` from the list of derived traits.

Reply:
That's a good point. Should we make the change in question incrementally and opportunistically, depending on how many applicable shapes there are? Once we cover enough shapes, letting them handle locally like above, maybe we can promote the logic to a centralized place? I wonder if that'll help keep the scope of this PR focused.

Reply:
I've understood the proposal better now. I'm going to create a new branch out of this, make the proposed change to `BaseSymbolMetadataProvider.[structureMeta | unionMeta | enumMeta]`, remove the above code from `EnumGenerator.kt` and `StructureGenerator.kt`, and run the tests in CI to see how many will break.

Reply:
Created #2029